src/object_store/wrapped.rs

```diff
@@ -252,3 +287,63 @@ impl ObjectStore for InternalObjectStore {
         self.inner.rename_if_not_exists(from, to).await
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::config::context::build_object_store;
+    use crate::config::schema::{ObjectStore, S3};
+    use crate::frontend::http::tests::deterministic_uuid;
+    use crate::object_store::wrapped::InternalObjectStore;
+    use datafusion::common::Result;
+    use deltalake::logstore::LogStore;
+    use rstest::rstest;
+
+    #[rstest]
+    #[case::bucket_root("test-bucket", "01020304-0506-4708-890a-0b0c0d0e0f10")]
+    #[case::path_no_delimiter(
+        "test-bucket/some/path/no/delimiter",
+        "some/path/no/delimiter/01020304-0506-4708-890a-0b0c0d0e0f10"
+    )]
+    #[case::path_with_delimiter(
+        "test-bucket/some/path/with/delimiter/",
+        "some/path/with/delimiter/01020304-0506-4708-890a-0b0c0d0e0f10"
+    )]
+    #[test]
+    fn test_table_location_s3(
+        #[case] bucket: &str,
+        #[case] table_prefix: &str,
+        #[values(None, Some("http://127.0.0.1:9000".to_string()))] endpoint: Option<
+            String,
+        >,
+    ) -> Result<()> {
+        let config = ObjectStore::S3(S3 {
+            region: None,
+            access_key_id: "access_key_id".to_string(),
+            secret_access_key: "secret_access_key".to_string(),
+            bucket: bucket.to_string(),
+            endpoint: endpoint.clone(),
+            cache_properties: None,
+        });
+        // In principle for this test we could use any object store since we only exercise the
+        // prefix/log store uri logic
```

Comment:
If you're worried about this, you could break out `table_prefix` into a separate function that operates on `(schema::ObjectStore, Url) -> Url` and test it without having to create an object store (don't know how much it's worth it).

Reply:
Not really worried, intended more like a clarification. The test also exercises the log store uri, which does need some object store.